src/control/system/database.go

```diff
@@ -130,18 +165,24 @@ func (sta *syncTCPAddr) get() *net.TCPAddr {
 }
 
 // NewDatabase returns a configured and initialized Database instance.
-func NewDatabase(log logging.Logger, cfg *DatabaseConfig) *Database {
+func NewDatabase(log logging.Logger, cfg *DatabaseConfig) (*Database, error) {
```

Comment:
Doesn't look like we're actually returning an error here yet. Plans to add some kind of validation here?